processing/src/main/java/io/druid/segment/data/CompressedIntsIndexedSupplier.java

```diff
@@ -0,0 +1,353 @@
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package io.druid.segment.data;
+
+import com.google.common.base.Preconditions;
+import com.google.common.io.Closeables;
+import com.google.common.primitives.Ints;
+import com.metamx.common.IAE;
+import com.metamx.common.guava.CloseQuietly;
+import io.druid.collections.ResourceHolder;
+import io.druid.collections.StupidResourceHolder;
+import io.druid.segment.CompressedPools;
+
+import java.io.IOException;
+import java.nio.ByteBuffer;
+import java.nio.ByteOrder;
+import java.nio.IntBuffer;
+import java.nio.channels.WritableByteChannel;
+import java.util.Iterator;
+import java.util.List;
+
+public class CompressedIntsIndexedSupplier implements WritableSupplier<IndexedInts>
+{
+  public static final byte version = 0x2;
+  public static final int MAX_INTS_IN_BUFFER = CompressedPools.BUFFER_SIZE / Ints.BYTES;
+
+
+  private final int totalSize;
+  private final int sizePer;
+  private final GenericIndexed<ResourceHolder<IntBuffer>> baseIntBuffers;
+  private final CompressedObjectStrategy.CompressionStrategy compression;
+
+  CompressedIntsIndexedSupplier(
+      int totalSize,
+      int sizePer,
+      GenericIndexed<ResourceHolder<IntBuffer>> baseIntBuffers,
+      CompressedObjectStrategy.CompressionStrategy compression
+  )
+  {
+    this.totalSize = totalSize;
+    this.sizePer = sizePer;
+    this.baseIntBuffers = baseIntBuffers;
+    this.compression = compression;
+  }
+
+  public int size()
+  {
+    return totalSize;
+  }
+
+  @Override
+  public IndexedInts get()
+  {
+    final int div = Integer.numberOfTrailingZeros(sizePer);
+    final int rem = sizePer - 1;
+    final boolean powerOf2 = sizePer == (1 << div);
+    if(powerOf2) {
+      return new CompressedIndexedInts() {
+        @Override
+        public int get(int index)
+        {
+          // optimize division and remainder for powers of 2
+          final int bufferNum = index >> div;
+
+          if (bufferNum != currIndex) {
+            loadBuffer(bufferNum);
+          }
+
+          final int bufferIndex = index & rem;
+          return buffer.get(buffer.position() + bufferIndex);
+        }
+      };
+    } else {
+      return new CompressedIndexedInts();
+    }
+  }
+
+  public long getSerializedSize()
+  {
+    return baseIntBuffers.getSerializedSize() + 1 + 4 + 4 + 1;
+  }
+
+  public void writeToChannel(WritableByteChannel channel) throws IOException
+  {
+    channel.write(ByteBuffer.wrap(new byte[]{version}));
+    channel.write(ByteBuffer.wrap(Ints.toByteArray(totalSize)));
```

Comment:
no, byteorder only applies to data, not the header. I think it makes sense to keep it consistent with all the other column types.